js_modules/dagit/packages/core/src/CapturedLogPanel.tsx

```diff
@@ -0,0 +1,96 @@
+import {Colors, Icon} from '@dagster-io/ui';
+import * as React from 'react';
+
+import {RawLogContent} from './RawLogContent';
+import {AppContext} from './app/AppContext';
+import {useCapturedLogs} from './useCapturedLogs';
+
+interface CapturedLogProps {
+  logKey: string[];
+  visibleIOType: string;
+  onSetDownloadUrl?: (url: string) => void;
+}
+
+interface CapturedOrExternalLogPanelProps extends CapturedLogProps {
+  externalUrl?: string;
+}
+
+export const CapturedOrExternalLogPanel: React.FC<CapturedOrExternalLogPanelProps> = React.memo(
+  ({externalUrl, ...props}) => {
+    if (externalUrl) {
+      return (
+        <div
```

Comment:
Minor nit but it'd be nice to use one of our Box components for this - I think:

```
<Box flex={{direction: 'row', alignItems: 'center', justifyContent: 'center', gap: 4}} background={Colors.Gray900} style={{color: Colors.White, minHeight: 0}}>
```

Would do it - the `gap` CSS property also lets you avoid having to put the marginleft + marginRight on the link